vignettes/baggr_model_convergence.Rmd

```diff
@@ -0,0 +1,119 @@
+
+# Warnings like "divergent transitions"
+
+The gradient is passed into a Hamiltonian equation, which simulates a path for where it thinks the posterior is headed. But this trajectory is based on the local behavior of the posterior, and depending on the geometry, if we move too far away from that point we might get more and more off base. Stan, on the back end, uses the `adapt_delta` parameter to determine how often it needs to check its simulated paths, re-evaluating the trajectory it is using.
+
+If you make `adapt_delta` really big, you end up moving really slowly, but you also are less likely to fly off the true trajectory of the posterior distribution. But if you make it small, you go faster, etc. You get the idea. How high you should set the parameter depends on the extent to which the algorithm is properly modeling the paths. So if you encounter divergent transitions, it is often good to raise `adapt_delta`. This discussion is really oversimplifying things to build the intuition, but you can find an (accessible) deeper discussion [here](https://arxiv.org/abs/1701.02434)
+if it is of interest!
+
+When you get a warning that there are a certain number of "divergent transitions," it means that the samples have gone far away from the anticipated path, and (at worse) are reduced to a random walk. This means that they may not be fully exploring the posterior distribution, and so you may need more iterations to get a good approximation of the distribution.
+
+If you get this warning and want to re-estimate your model with a higher `adapt_delta` setting, you can change the argument in the call to the `baggr` function like so:
+
+```{r}
+# the ... parameters in the call to baggr() are
+# passed to rstan::sampling
+# for documentation see ?rstan::sampling
+fit <- baggr(schools, model = "rubin",
+      pooling = "partial",
+      refresh = 0, # silence printing MCMC draws
+      prior_hypermean = normal(0, 10),
+      prior_hypersd = cauchy(0, 3),
+      control = list(
+        adapt_delta = 0.9 # 0.8 by default
+      ))
+
+fit
+```
+
+If you still get a lot of divergent transitions after raising the `adapt_delta` really high, it may indicate an issue with your model! This is what is commonly referred to as the [folk theorem of statistical computing](https://statmodeling.stat.columbia.edu/2008/05/13/the_folk_theore/).
+
+# $\widehat{R}$ being too low
+
+The other warning you get, that “r hat” statistics are too high, indicates that the effective sample size (ESS) of your draws are too low and your chains have not converged. Because MCMC is not a deterministic algorithm, we run multiple chains of samples in order to check that each has converged to a good enough answer.
```

Comment:
These are 2 mutual issues? ie correlated, but can have low R hat, low ESS, no?

Reply:
Yes you are right, this is misleading at best, and at worst incorrect.